File: rb/features.rb
# Navalnyarchive SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module NavalnyarchiveFeatures
  def self.make_feature(name)
    case name
    when "base"
      NavalnyarchiveBaseFeature.new
    when "test"
      NavalnyarchiveTestFeature.new
    else
      NavalnyarchiveBaseFeature.new
    end
  end
end

File: py/features.py
# Navalnyarchive SDK feature factory

from feature.base_feature import NavalnyarchiveBaseFeature
from feature.test_feature import NavalnyarchiveTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NavalnyarchiveBaseFeature(),
        "test": lambda: NavalnyarchiveTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
